lib/src/mapbox_maps_platform.dart

```diff
@@ -181,3 +187,36 @@ class _MapboxMapsPlatform {
     }
   }
 }
+
+/// A registry to hold suffixes for Channels.
+///
+class _SuffixesRegistry {
+  _SuffixesRegistry._instance();
+
+  int _suffix = -1;
+  final Set<int> suffixesInUse = {};
+  final Set<int> suffixesAvailable = {};
+
+  int getSuffix() {
+    int suffix;
+
+    if (suffixesAvailable.isEmpty) {
+      _suffix++;
+      suffixesInUse.add(_suffix);
+      suffix = _suffix;
+    } else {
+      suffix = suffixesAvailable.first;
+      suffixesAvailable.remove(suffix);
+      suffixesInUse.add(suffix);
+    }
+
+    return suffix;
+  }
+
+  void releaseSuffix(int suffix) {
+    if (!suffixesInUse.remove(suffix)) {
+      throw Exception("Suffix $suffix not in use");
```

Comment:
imo pretty weird to have an exception there, generally this should not happen at all

Reply:
Should never happen, added to test the algo. Will remove